### ISSUE TITLE
False positives when redefining property in subclass

### ISSUE THREAD

**Issue by mthuurne** [python/mypy#5936]
I'm running mypy on the following code: (no additional arguments)
```python
class Base:
	def __init__(self, value):
		self._value = value

	@property
	def value(self):
		return self._value

class Sub(Base):
	@Base.value.setter
	def value(self, value):
		self._value = value
```
And it reports:
```
inherited_property.py:10: error: "Callable[[Any], Any]" has no attribute "setter"
```
Line 10 is `@Base.value.setter`, so it seems to have a problem with referring to a property defined in a different class. Python itself has no problem with this though.

If the property has a setter in the base class, a slightly different error is reported:
```python
class Base:
	def __init__(self, value):
		self._value = value

	@property
	def value(self):
		return self._value

	@value.setter
	def value(self, value):
		self._value = value

class Sub(Base):
	@Base.value.setter
	def value(self, value):
		self._value = value
```
This results in mypy reporting:
```
inherited_property.py:14: error: overloaded function has no attribute "setter"
```
Again the error is reported on the `@Base.value.setter` line.

I'm using a mypy dev snapshot (0.650+) on Python 3.6.5.

I found the existing issue #220, but that is so generic that I figured it would make sense to open a new separate issue for tracking this specific use case.

**Comment by JukkaL** [comment]
Yeah, this is something that mypy should perhaps support. This is the first time I've seen this idiom, so it's probably not very common though. 

This workaround works currently:

```py
class Base:
	def __init__(self, value: int) -> None:
		self._value = value

	@property
	def value(self) -> int:
		return self._value

class Sub(Base):
	@property
	def value(self) -> int:
		return super().value

	@value.setter
	def value(self, value: int) -> None:
		self._value = value
```

**Comment by mthuurne** [comment]
This problem might be a special case of #4644.

**Comment by mthuurne** [comment]
Actually it looks like both are a special case of #1465, as pointed out in [this comment](https://github.com/python/mypy/issues/1465#issuecomment-384825082).

Changing `@value.setter` to `@Base.value.setter` in your example is accepted by mypy. So it seems the problem is not the `Base.value` idiom, but the fact that mypy doesn't recognize a property setter unless it is directly preceded by the getter.

**Comment by dmanser92** [comment]
I had the same issue. Because I have a lot of sub-classes and I don't want to copy the property around so many times I ended up doing this:
- added a type-ignore comment to the decorators where I overwrite the setters, like `@Base.value.setter  # type: ignore[attr-defined]`
- made a comment in the base-class explaining why I do this and pointing to this post

**Comment by Cleptomania** [comment]
I've run in to this issue as well and created #14684, and I've found another work-around that produces no errors from MyPy. It seems to not be problematic if the property in the base class is declared using the `property()` function as opposed to the decorators, like below, note that my specific use case was attempting to only override the setter of the Base class.

```python
class Base:
    def __init__(self, value: int) -> None:
        self._value = value

    def _get_value(self) -> int:
        return self._value

    def _set_value(self, value: int):
        self._value = value

    value = property(_get_value, _set_value)
    
class Sub(Base):
    def __init__(self, value: int) -> None:
        super().__init__(value)
        
    @Base.value.setter
    def value(self, value: int) -> None:
        Base.value.fset(self, value)
        print("Do Something after the Base setter")
```
